Cql/Cql.Firely/Extensions/CqlComparersExtensions.cs

```diff
@@ -45,44 +45,23 @@ public static CqlComparers AddFhirComparers(this CqlComparers comparers)
             comparers.Register(typeof(Uuid), new IValueComparer<string?>());
             comparers.Register(typeof(Identifier), new IdentifierComparer(comparers, comparers));
 
-            comparers.Register(typeof(Code<>), (type, _comparers) =>
+            comparers.Register(typeof(Code<>), (type, self) =>
             {
-                var codeType = type.GetGenericArguments()[0];
-                var comparerType = typeof(CodeComparer<>).MakeGenericType(codeType);
-                var codeComparer = (ICqlComparer)Activator.CreateInstance(comparerType, _comparers)!;
-                var codeStringComparer = new CodeStringComparer(codeComparer);
-                return codeStringComparer;
+                Type codeType = type.GetGenericArguments()[0];
+                Trace.Assert(codeType.IsEnum());
+                Type comparerType = typeof(CodeOfEnumCqlComparer<>).MakeGenericType(codeType);
+                ICqlComparer codeOfEnumCqlComparerInstance = (ICqlComparer)comparerType.InvokeMember(
+                    nameof(CodeOfEnumCqlComparer<DummyEnum>.Instance),
+                    BindingFlags.Static | BindingFlags.Public | BindingFlags.GetProperty,
+                    null, null, [])!;
+                PrimitiveTypeAgainstStringComparer primitiveTypeAgainstStringComparer = new PrimitiveTypeAgainstStringComparer(codeOfEnumCqlComparerInstance);
+                return primitiveTypeAgainstStringComparer;
             });
 
             return comparers;
         }
 
-        private class CodeStringComparer(ICqlComparer inner) : CqlComparerDecorator(inner)
-        {
-            public override int? Compare(
-                object? x,
-                object? y,
-                string? precision)
-            {
-                // We always expect x to be a Code<T> but we only need the ObjectValue on the non-generic base type PrimitiveType.
-                if (x is PrimitiveType xCode && y is string yString)
-                {
-                    if (precision != null)
-                        throw new InvalidOperationException(
-                            $"Precision '{precision}' is not supported for comparing Code<T> to string.");
-
-                    return StringComparer.Ordinal.Compare(xCode.ObjectValue, yString);
-                }
-
-                return base.Compare(x, y, precision);
-            }
-
-            public override bool? Equals(
-                object? x,
-                object? y,
-                string? precision) =>
-                throw new UnreachableException("CqlComparers always goes through Compare, so we never reach here");
-        }
+        private enum DummyEnum {}
```

Comment:
I think it's worth a few lines of documentation to describe what you use this for.